doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,328 @@
+
+ - rdm-draft-petry-riot-design-goals.md
+ - Title: RIOT design goals
+ - Author: Daniel Petry
+ - Date: October 2018
+
+## Abstract
+
+This document presents key design aspects of the RIOT operating system: its
+application contexts and design rationales.  It aims to reflect a consensus on
+the reasoning behind RIOT design, in order to help new contributors get up to
+speed more quickly and provide common ground for technical debates.
+
+## Status
+
+The content of this document is licensed with a Creative Commons CC-BY-SA
+license.
+
+## Terminology
+
+Throughout this document, the term "users" refers to developers who are using
+RIOT as a basis to implement application software, without editing RIOT's
+source code. The term "developers" refers to contributors to the source code of
+RIOT.
+
+# 1. Introduction
+
+The RIOT developer community has grown from a state of a few developers in
+relatively close contact to a highly distributed worldwide organisation with
+members joining continuously. As a result, passing on underlying understanding
+and assumptions that drive design decisions via word-of-mouth is no longer
+feasible. 
+
+This document represents the consensus on a set of generalised, loose
+requirements for RIOT (in a voluntary open source context, gathering strict
+requirements is neither possible nor welcome).
+
+The first section discusses some of the concrete [use cases](#2-use-cases)
+which RIOT is driven by, to give a broad awareness to the developer where and
+how their features will be used. Based on this bigger picture, the [design
+philosophies](#3-design-philosophies) in the second section provide specific
+guidance for design. These include a focus on constrained devices, a short
+learning curve, and the versatility to support a huge variety of devices and
+functionality.
+
+# 2. Use cases
+
+RIOT is a general purpose IoT operating system for low-end devices, such as
+those described in [1]. These devices have a low memory footprint - kilobytes
+rather than megabytes - and an extremely low memory footprint of a few
+kilobytes in certain applications. As such, RIOT targets separate use cases from
```

Comment:
@danpetry seems like a copy/paste issue here (twice mentioning memory in the sentence?).

I propose rephrasing:

```
These devices have an extremely low memory footprint - kilobytes rather than megabytes or gigabytes - and are typically based on low-power microcontrollers lacking hardware features provided on less constrained devices e.g. an MMU. 
```

Reply:
This wasn't a copy/paste error: the idea was to say
a) our targeted devices all have ~kilobytes of memory
b) our targeted devices go down to a very small number of kilobytes in certain cases
Does that clarify and do you agree?

Reply:
Have clarified to:
These devices have a low memory footprint of kilobytes rather than megabytes, going down to a few kilobytes in certain cases.
Does that read better and make the meaning more apparent?